### PR TITLE
Preserve fenced and indented blocks after headings with --wrap

### LINKED ISSUE #261: `--wrap` joins lines inside fenced code blocks
## Summary

`mdtablefix --wrap` treats the content of fenced code blocks as prose
paragraphs, joining separate lines and re-wrapping them at the 80-column limit.
Code blocks should be preserved verbatim.

## Version

mdtablefix 0.3.1

## Replication

Create a file `repro.md` with the following content:

````markdown
## Verification

```bash
set -o pipefail
make check-fmt 2>&1 | tee /tmp/fmt.log
make lint 2>&1 | tee /tmp/lint.log
make test 2>&1 | tee /tmp/test.log
```
````

Run:

```bash
mdtablefix --wrap --in-place repro.md
```

### Expected output

The code block is unchanged — each shell command remains on its own line:

````markdown
## Verification

```bash
set -o pipefail
make check-fmt 2>&1 | tee /tmp/fmt.log
make lint 2>&1 | tee /tmp/lint.log
make test 2>&1 | tee /tmp/test.log
```
````

### Actual output

All lines inside the code block are joined into a single paragraph and
re-wrapped at 80 columns:

````markdown
## Verification

```bash
set -o pipefail make check-fmt 2>&1 | tee /tmp/fmt.log make lint 2>&1 | tee
/tmp/lint.log make test 2>&1 | tee /tmp/test.log
```
````

The resulting content is syntactically invalid bash — `set -o pipefail` receives
`make`, `check-fmt`, etc. as positional parameters instead of executing them as
separate commands.

## Regression test case

The following captures the invariant that fenced code blocks are not modified by
`--wrap`:

```text
Input:
    ```bash
    set -o pipefail
    make check-fmt 2>&1 | tee /tmp/fmt.log
    make lint 2>&1 | tee /tmp/lint.log
    make test 2>&1 | tee /tmp/test.log
    ```

Assert: output == input (byte-identical)
```

## Notes

- The `--wrap` flag's help text says "Wrap paragraphs and list items to 80
  columns", which implies code blocks are out of scope.
- Indented code blocks (four-space prefix) appear to be affected in the same
  way.
- The bug is independent of the code block's info string (`bash`, `text`, or
  none).

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Fix a bug where wrapping Markdown with --wrap could corrupt fenced code blocks, especially after headings. Also preserves indented (four-space) shell blocks following headings.
- Adds regression tests for both CLI and library wrap behavior to ensure fenced and indented blocks remain byte-identical after wrapping.

### Core Functionality
- Improve wrap logic to preserve fenced code blocks and indented shell blocks verbatim without altering their content.

### Tests
- New: tests/wrap_cli.rs — CLI regression tests ensuring `--wrap --in-place` leaves fenced and indented shell blocks verbatim after a heading.
- New: tests/wrap_unit.rs — unit tests including `wrap_text_preserves_fenced_shell_block_after_heading` and `wrap_text_preserves_indented_shell_block_after_heading` verifying that wrapping at 80 columns does not modify these blocks.

### Test plan
- Run unit tests: `cargo test`.
- Run CLI tests: `cargo test --tests` (or run the mdtablefix CLI regression suite).
- Specifically verify that fenced and indented code blocks are preserved byte-identically after wrapping.

◳ Generated by [DevBoxer](https://www.devboxer.com) ◰

---

ℹ️ Tag @devboxerhub to ask questions and address PR feedback

📎 **Task**: https://www.devboxer.com/task/5d86b2fd-6326-4240-a4a0-346e3af9237c

📝 Closes #261

## Summary by Sourcery

Add regression coverage to ensure Markdown wrapping preserves verbatim code blocks following headings.

Tests:
- Add unit tests verifying wrap_text preserves fenced and indented shell code blocks after a heading at 80 columns.
- Add CLI regression tests ensuring `mdtablefix --wrap --in-place` leaves fenced and indented shell code blocks byte-identical after a heading.

### CODE REVIEW

File: tests/wrap_unit.rs [hunk 1]
```diff
@@ -104,3 +104,33 @@ fn wrap_text_preserves_links() {
...
+        "## Verification".to_string(),
+        String::new(),
+        "    set -o pipefail".to_string(),
+        "    make check-fmt 2>&1 | tee /tmp/fmt.log".to_string(),
+        "    make lint 2>&1 | tee /tmp/lint.log".to_string(),
+        "    make test 2>&1 | tee /tmp/test.log".to_string(),
+    ];
+
+    assert_eq!(wrap_text(&input, 80), input);
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/wrap_unit.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:

## Overall Comments
- The two CLI regression tests share identical temp-file setup and invocation of `mdtablefix`; consider extracting that into a small helper to reduce duplication and make it easier to add future wrap-related cases.
- To harden the regression coverage, consider adding a variant of the tests where the input does not end with a trailing newline (and/or includes content after the code block) to ensure `--wrap --in-place` preserves code blocks and surrounding whitespace in those edge cases as well.

## Individual Comments

### Comment 1
<location path="tests/wrap_unit.rs" line_range="109-118" />
<code_context>
 }
+
+#[test]
+fn wrap_text_preserves_fenced_shell_block_after_heading() {
+    let input = vec![
+        "## Verification".to_string(),
+        String::new(),
+        "```bash".to_string(),
+        "set -o pipefail".to_string(),
+        "make check-fmt 2>&1 | tee /tmp/fmt.log".to_string(),
+        "make lint 2>&1 | tee /tmp/lint.log".to_string(),
+        "make test 2>&1 | tee /tmp/test.log".to_string(),
+        "```".to_string(),
+    ];
+
+    assert_eq!(wrap_text(&input, 80), input);
+}
+
</code_context>
<issue_to_address>
**suggestion (testing):** Consider adding a variant without a blank line between the heading and the fenced block

Wrapping around fenced blocks can differ when there’s no blank line after the heading. To better cover #261, please add a similar test without the empty string between the heading and the fence:

```rust
let input = vec![
    "## Verification".to_string(),
    "```bash".to_string(),
    // ... body ...
    "```".to_string(),
];
assert_eq!(wrap_text(&input, 80), input);
```
</issue_to_address>
~~~
